crates/ruff_python_ast/src/verbatim.rs

```diff
@@ -0,0 +1,1325 @@
+//! An equivalent object hierarchy to the `RustPython` AST hierarchy, but with an additional
+//! `Verbatim` node type that represents a verbatim string of Python code. Used to generate
+//! source code from the AST.
```

Comment:
This is basically copied from `comparable.rs` where we have a similar hierarchy but for creating hashable AST nodes. The difference is that this adds `ExprVerbatim` and `StmtVerbatim` to the hierarchy, and removes some things that don't matter for formatting (like `ExprContext`).